### ISSUE TITLE
tB ActiveX DLL slow in tB 

### ISSUE THREAD

**Issue by KarlSevenSeven** [WaynePhillipsEA/twinbasic#609]
**Describe the bug**
ActiveX DLL runs slower in tB than in VB6, it's half as fast.
Also there is a memory problem that comes sooner in tB.

**To Reproduce**
Create ActiveX DLL in tB from this:
[Fast DLL.zip](https://github.com/WaynePhillipsEA/twinbasic/files/7786422/Fast.DLL.zip)

```
Class c_Fast

	Public Function FastReplace(ByRef All As String, ByRef ReplaceWhat As String, ByRef ReplaceWith As String) As String
		FastReplace = Replace(All, ReplaceWhat, ReplaceWith)
	End Function
	
	Public Function FastSplit(ByVal Expression As String, Optional ByRef Delimiter As Variant, Optional ByVal Limit As Long = -1&, Optional ByVal Compare As VbCompareMethod = 1&) As Variant
		FastSplit = Split(Expression, Delimiter, Limit, Compare)
	End Function
	
End Class
```
Run a test app that references the DLL.
[TestApp.zip](https://github.com/WaynePhillipsEA/twinbasic/files/7786399/TestApp.zip)
In the 'build large string' section, change 7 to 6 for the first few runs.

**Expected behavior**
Same performance as in VB

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - twinBASIC 0.13.59

**Additional context**
I wanted to test if the nice performance of some tB functions do their job in a VB6 app.
In general this seems to work.
To be sure, I ran the same test code in tB.

**Comment by WaynePhillipsEA** [comment]
I won't be able to look at this until next week.   If I understood correctly, the compiled DLL works as expected when the methods are called from VB6, but the slow down appears when calling the same methods from another tB project.  Is that correct?

**Comment by KarlSevenSeven** [comment]
Correct.

**Comment by KarlSevenSeven** [comment]
Having a better example now.

**To Reproduce:**
1)
Compile the DLL with VS running ad admin from this:
[tB DLL.zip](https://github.com/WaynePhillipsEA/twinbasic/files/7790154/tB.DLL.zip)
2)
Close VS and start it as normal user.
Load this test app and adjust reference to the compiled DLL:
[tB TestApp.zip](https://github.com/WaynePhillipsEA/twinbasic/files/7790161/tB.TestApp.zip)
3)
Let it work and see numbers like this:
```
Executing 'Test.m_misc.Main'...
build $: 32 msec.
101 dummy calls: 14093 msec.

```
4)
Load this test VB6 test app, adjust reference to the compiled DLL from step 1):
[VB6 TestApp.zip](https://github.com/WaynePhillipsEA/twinbasic/files/7790170/VB6.TestApp.zip)
Let it work and see numbers like this:
```
build $: 32 msec.
101 dummy calls: 2094 msec.
```
---

In step 3), at least seen in the task manager, the memory usage is quite high.

---

If you increase the test count to 1000, you'll get such timings:
```
tB 141188 msec.
VB6  21515 msec.
```
So the factor is quite constant.

**Comment by KDGundermann** [comment]
I'm trying to reproduce this on my Pc.
I'm getting a 
![image](https://user-images.githubusercontent.com/208315/148650683-0542303d-cdef-423d-91a6-a063314a8cfc.png)
when building..

**Comment by WaynePhillipsEA** [comment]
This slowdown is due to us not populating the ThreadingModel in the registry for the class.  By not providing a ThreadingModel value, object calls on the COM object (c_Fast) are being marshalled to the root STA thread, causing much extra overhead in marshalling the large string.

This bug is only currently affecting the code when run under the debugger, since the tB debugger doesn't run code on the root STA thread.  When you build to an EXE, the code naturally runs on the main thread which is of course the root STA thread and so no marshalling occurs.

VB6 is not affected by this (even when debugging), because it always runs code on the root STA thread, and so no marshalling occurs.

The fix is easy; we just need to populate the ThreadingModel in the registry when registering our components.  This will be fixed today.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.13.69.  You'll need to recompile the DLL with v0.13.69.   Thanks!
